test/functional/apps/dashboard/panel_context_menu.ts

```diff
@@ -0,0 +1,191 @@
+      await PageObjects.dashboard.initTests();
+      await PageObjects.dashboard.preserveCrossAppState();
+      await PageObjects.dashboard.clickNewDashboard();
+      await PageObjects.timePicker.setHistoricalDataRange();
+      await dashboardAddPanel.addVisualization(PIE_CHART_VIS_NAME);
+    });
+
+    after(async function () {
+      await PageObjects.dashboard.gotoDashboardLandingPage();
+    });
+
+    it('are hidden in view mode', async function () {
+      await PageObjects.dashboard.saveDashboard(dashboardName);
+
+      await dashboardPanelActions.openContextMenu();
+      await dashboardPanelActions.expectMissingEditPanelAction();
+      await dashboardPanelActions.expectMissingRemovePanelAction();
+    });
+
+    it('are shown in edit mode', async function () {
+      await PageObjects.dashboard.switchToEditMode();
+
+      const isContextMenuIconVisible = await dashboardPanelActions.isContextMenuIconVisible();
+      expect(isContextMenuIconVisible).to.equal(true);
+
+      await dashboardPanelActions.expectExistsEditPanelAction();
+      await dashboardPanelActions.expectExistsClonePanelAction();
+      await dashboardPanelActions.expectExistsReplacePanelAction();
+      await dashboardPanelActions.expectExistsRemovePanelAction();
+    });
+
+    it('are shown in edit mode after a hard refresh', async () => {
+      // Based off an actual bug encountered in a PR where a hard refresh in
+      // edit mode did not show the edit mode controls.
+      const currentUrl = await browser.getCurrentUrl();
+      // The second parameter of true will include the timestamp in the url and
+      // trigger a hard refresh.
+      await browser.get(currentUrl.toString(), true);
+      await PageObjects.header.waitUntilLoadingHasFinished();
+
+      await dashboardPanelActions.expectExistsEditPanelAction();
+      await dashboardPanelActions.expectExistsClonePanelAction();
+      await dashboardPanelActions.expectExistsReplacePanelAction();
+      await dashboardPanelActions.expectExistsRemovePanelAction();
+
+      // Get rid of the timestamp in the url.
+      await browser.get(currentUrl.toString(), false);
+    });
+
+    describe('visualization object edit menu', () => {
+      it('opens a visualization when edit link is clicked', async () => {
+        await dashboardPanelActions.clickEdit();
+        await PageObjects.header.waitUntilLoadingHasFinished();
+        const currentUrl = await browser.getCurrentUrl();
+        expect(currentUrl).to.contain(VisualizeConstants.EDIT_PATH);
+      });
+
+      it('deletes the visualization when delete link is clicked', async () => {
+        await PageObjects.header.clickDashboard();
+        await PageObjects.header.waitUntilLoadingHasFinished();
+        await dashboardPanelActions.removePanel();
+
+        const panelCount = await PageObjects.dashboard.getPanelCount();
+        expect(panelCount).to.be(0);
+      });
+    });
+
+    /*
+    describe('on an expanded panel', function () {
```

Comment:
why is this commented out?

Reply:
Test test suite was broken, so I temporarily commented it out to figure out the reason and fix it. I've figured it out and now it is back uncommented.